### PR TITLE
Upgrade GitHub Actions

### PR DESCRIPTION
**Pull request by cclauss**
https://github.com/nodejs/release#readme
https://github.com/actions/checkout/releases
https://github.com/actions/setup-node/releases
https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
> CI Always set to true.

## :recycle: Current situation

*Describe the current situation. Explain current problems, if there are any. Be as descriptive as possible (e.g., including examples or code snippets).*

## :bulb: Proposed solution

*Describe the proposed solution and changes. How does it affect the project? How does it affect the internal structure (e.g., refactorings)?*

## :gear: Release Notes

*Provide a summary of the changes or features from a user's point of view. If there are breaking changes, provide migration guides using code examples of the affected features.*

## :heavy_plus_sign: Additional Information
*If applicable, provide additional context in this section.*

### Testing

*Which tests were added? Which existing tests were adapted/changed? Which situations are covered, and what edge cases are missing?*

### Reviewer Nudging

*Where should the reviewer start? what is a good entry point?*

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -27,5 +27,3 @@ jobs:
 
       - name: Build the project
         run: npm run build
-        env:
```
Comment (bwp91) [inline .github/workflows/build.yml:30]: what does this change do?

**Comment by bwp91** [comment]
Hi @cclauss  looking into these PRs now, please rebase on `main` which i renamed from `master`

**Comment by bwp91** [comment]
Closing this as I don't believe is needed anymore, thanks for your contributions @cclauss
